active/teachingtip/TeachingTipSpec.md

```diff
@@ -0,0 +1,591 @@
+                <p>You can change your tip preferences in <a href="app:/item/SettingsPage">Settings</a> if you change your mind.</p>
+        </muxc:TeachingTip>
+    </muxc:TeachingTip.Attach>
+</Button>
+```
+
+C#
+```C#
+public void OnFirstSaveButtonClick(object sender, RoutedEventArgs args)
+{
+    TeachingTipExample.IsOpen=true;
+}
+```
+
+<a href="url"><img src="TeachingTipButtonsSample.jpg" height="600"></a>
+
+### Hero content
+
+Edge to edge content can be added to a teaching tip by setting the HeroContent property. The location of hero content can be set to the top or bottom of a teaching tip by setting the HeroContentPlacement property.
+
+XAML
+```XAML
+<Button Content="Tip Example" Click="OnFirstSaveButtonClick">
+    <muxc:TeachingTip.Attach> 
+        <muxc:TeachingTip x:Name="AutoSaveTip"
+            Title="Saving automatically"
+            Subtitle="We save your changes as you go - so you never have to."
+            <muxc:TeachingTip.HeroContent>
+                <Image Source="Assets/Giraffe.png" />
+            </muxc:TeachingTip.HeroContent>
+            Margin="15">
+        </muxc:TeachingTip>
+    </muxc:TeachingTip.Attach>
+</Button>
+```
+
+C#
+```C#
+public void OnFirstSaveButtonClick(object sender, RoutedEventArgs args)
+{
+    TeachingTipExample.IsOpen=true;
+}
+```
+
+<a href="url"><img src="TeachingTipHeroContentSample.jpg" height="600"></a>
+
+### Add an icon
+
+An icon can be added to the side of the title and subtitle using the IconSource property.
+
+XAML
+```XAML
+<Button Content="Tip Example" Click="OnFirstSaveButtonClick">
+    <muxc:TeachingTip.Attach> 
+        <muxc:TeachingTip x:Name="AutoSaveTip"
+            Title="Saving automatically"
+            Subtitle="We save your changes as you go - so you never have to."
+            IconSource="SaveIcon.png"
+            Margin="15">
+        </muxc:TeachingTip>
+    </muxc:TeachingTip.Attach>
+</Button>
+```
+
+C#
+```C#
+public void OnFirstSaveButtonClick(object sender, RoutedEventArgs args)
+{
+    TeachingTipExample.IsOpen=true;
+}
+```
+
+<a href="url"><img src="TeachingTipIconSample.jpg" height="600"></a>
+
+### Enable light-dismiss
+
+A teaching tip may be light-dismiss enabled so that it will dismiss when a user scrolls or interacts with other elements of the application.
+
+XAML
+```XAML
+<Button Content="Tip Example" Click="OnFirstSaveButtonClick">
+    <muxc:TeachingTip.Attach> 
+        <muxc:TeachingTip x:Name="AutoSaveTip"
+            Title="Saving automatically"
+            Subtitle="We save your changes as you go - so you never have to."
+            IsLightDismissEnabled="True"
+            Margin="15">
+        </muxc:TeachingTip>
+    </muxc:TeachingTip.Attach>
+</Button>
+```
+
+C#
+```C#
+public void OnFirstSaveButtonClick(object sender, RoutedEventArgs args)
+{
+    TeachingTipExample.IsOpen=true;
+}
+```
+
+<a href="url"><img src="TeachingTipLightDismissSample.jpg" height="600"></a>
```

Comment:
The screenshot implies that no close buttons are shown when light dismiss is enabled. Is that true? If so, maybe worth calling out in the doc? And I assume an app developer still can show a close button by setting appropriate properties, right?

Reply:
This is correct and I have updated the "Enable light-dismiss" example description to call this behavior out. 

As of now, the developer cannot override this. The absence of a close button is used to signal light-dismiss behavior to users. User testing found that that including close buttons on light-dismiss tips was a source of frustration for users as users would expect these tips to **not** behave like light-dismiss tips only to have them disappear if they were not the next thing interacted with. For consistency, tips are either light-dismiss or manual click dismiss.

Reply:
This is correct and I have updated the "Enable light-dismiss" example description to call this behavior out.

As of now, the developer cannot override this. The absence of a close button is used to signal light-dismiss behavior to users. User testing found that that including close buttons on light-dismiss tips was a source of frustration for users as users would expect these tips to not behave like light-dismiss tips only to have them disappear if they were not the next thing interacted with. For consistency, tips are either light-dismiss or manual click dismiss.

Reply:
`Escape` should close the tip for keyboard accessibility, eh?

Reply:
@michael-hawker, correct! This behavior is outlined in the Appendix > Design Notes > Accessibility section.